patroni/postgresql/slots.py

```diff
@@ -173,13 +247,24 @@ def process_permanent_slots(self, slots: List[Dict[str, Any]]) -> Dict[str, int]
         return ret
 
     def load_replication_slots(self) -> None:
+        """Query replication slot information from the database and store it for processing by other tasks.
+
+        .. note::
+            Only supported from PostgreSQL version 9.4 onwards.
+            Runs when scheduled by HA loop.
+
+        Store replication slot ``name``, ``type``, ``plugin``, ``database`` and ``datoid``.
+        If PostgreSQL version is 10 or newer also store ``catalog_xmin`` and ``confirmed_flush_lsn``.
+
+        When using logical slots, store information separately for slot synchronisation (copying) on replica nodes.
```

Comment:
It is not used for copying.
Logical slots could already exist, but we have to check that logical slot will be safe to use after promotion.
The actual check is executed in the `_ready_logical_slots()` method, which removes keys from `self._logical_slots_processing_queue` when the slot is fine.

Reply:
```suggestion
        When using logical slots, store information separately for slot synchronisation  on replica nodes.
```